api/tokens.go

```diff
@@ -165,3 +219,74 @@ func handleUnexpectedError(ctx context.Context, w http.ResponseWriter, err error
 	writeErrorResponse(ctx, w, http.StatusInternalServerError, errorResponseBody)
 	return
 }
+
+func computeSecretHash(clientSecret string, username string, clientId string) string {
```

Comment:
This may want to be refactored to a more shared location at some point as other endpoints will need it

Reply:
Yes good idea, we have created a utility package